packages/block-editor/src/hooks/duotone.js

```diff
@@ -304,28 +297,63 @@ function BlockDuotoneStyles( { name, duotoneStyle, id } ) {
  */
 const withDuotoneStyles = createHigherOrderComponent(
 	( BlockListBlock ) => ( props ) => {
-		const duotoneSupport = getBlockSupport(
-			props.name,
-			'color.__experimentalDuotone'
-		);
+		const id = useInstanceId( BlockListBlock );
+
+		const { selector, support } = useMemo( () => {
```

Comment:
I figured this was worth memoizing because there were some performance issues noted in #48401